### ISSUE TITLE
A11Y: Section expand/collapse in design panel

### ISSUE THREAD

**Issue by dreamofabear** [GoogleForCreators/web-stories-wp#4603]
## Bug Description

For sections "Size & position", "Layer", "Style", and "Link" in the design panel:

1. Mouse users can click anywhere in a design panel section header to toggle expanded/collapsed state. Keyboard users, however, must navigate into A11Y subtree of the `<h2>` and select and activate the `v` or `^` button.
2. The button changes its A11Y name based on expanded/collapsed state instead of using `aria-expanded`.

## Expected Behavior

1. Announce each section header as a button that toggles expansion/collapse on click. Disable each header's A11Y subtree (don't allow navigating into children).
2. Add `aria-expanded` to each section header to convey expanded/collapsed state. 

## Steps to Reproduce

Enable VoiceOver, navigate into the design panel, and highlight a section header.

## Additional Context

- Plugin Version: 1.0.0
- Operating System: macOS 10.15.6
- Browser: Chrome 85

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by zidianlyu** [comment]
@choumx to confirm, do we need to disable the a11y accessibility on the `Border` section?

This section:
![Screen Shot 2020-11-12 at 7 52 30 AM](https://user-images.githubusercontent.com/11252886/98963076-1f267e00-24bc-11eb-8356-50a867c015b1.png)

Also, may I have an example/steps to bring up the `Style` section, I tried images and videos, but neither of them brings up a `Style` section?

**Comment by zidianlyu** [comment]
@choumx for the `Style` section, do you mean the section for the color style?

This section:
![Screen Shot 2020-11-12 at 8 02 08 AM](https://user-images.githubusercontent.com/11252886/98964262-74af5a80-24bd-11eb-9137-34449409be8d.png)

**Comment by zidianlyu** [comment]
I got all those 4 sections.

**Comment by barklund** [comment]
@choumx I'm not sure I understand the purpose of item 1 in this. Also, the implementation seems to completely miss the point here. The accessibility subtree of the header is to be disabled, not [the entire panel](https://github.com/google/web-stories-wp/pull/5263/files)!

And why only those 4 panels? All the panel headers work in the same way, no?

**Comment by dreamofabear** [comment]
>  I'm not sure I understand the purpose of item 1 in this.

Can you elaborate?

> Also, the implementation seems to completely miss the point here. The accessibility subtree of the header is to be disabled, not the entire panel!

Yea, the first implementation was way off. Discussed with @zidianlyu offline.

> And why only those 4 panels? All the panel headers work in the same way, no?

Yep we should fix all panels. I think those 4 were the ones reported by the A11Y review.

**Comment by barklund** [comment]
> > I'm not sure I understand the purpose of item 1 in this.
> 
> Can you elaborate?

I don't understand the issue with this item in the ticket description:

> 1. Mouse users can click anywhere in a design panel section header to toggle expanded/collapsed state. Keyboard users, however, must navigate into A11Y subtree of the `<h2>` and select and activate the `v` or `^` button.

Keyboard users can't navigate directly to the H2 anyway (using tab at least) and if pressing tab before or shift-tab after the headline, focus will move to the button collapsing/expanding the section. Is there another way of navigating as a keyboard user, where you can navigate to the H2 and not directly to the button inside the H2?

**Comment by dreamofabear** [comment]
Hmm, are you using a screen reader? 

This bug was reproduced with VoiceOver on macOS. See "Steps to reproduce":
> Enable VoiceOver, navigate into the design panel, and highlight a section header.

**Comment by barklund** [comment]
> Hmm, are you using a screen reader?

Tab order is the same with and without a screenreader. And yes, I test with VoiceOver on mac.

> This bug was reproduced with VoiceOver on macOS. See "Steps to reproduce":
> 
> > Enable VoiceOver, navigate into the design panel, and highlight a section header.

What does "highlight" mean? Unfortunately, I can't both record my screen and use VoiceOver, as the two things conflict with each other and VoiceOver keep talking about my screen recording rather than what I'm doing, but I cannot recreate this.

The steps to recreate doesn't even specify **what** to recreate. What is the bug? What is happening that is wrong?

Please detail the steps further (with actual shortcuts or buttons pressed) and explain what is observed and what should happen instead, so I can reproduce this locally.

I have no problems using VoiceOver while navigating to and from the different panels. Yes, changing the label of the panel expand/collapse button to just say "Layer panel" (rather than "expand/collapse panel") makes sense (as the current expanded/collapsed state is already announced), but other than that I don't have any issues.

Which is why I am asking: What is the problem here? I don't understand the issue.

**Comment by dreamofabear** [comment]
Actual:
1. Sections (`<Header>`) are not announced as buttons
2. Sections change ARIA label to "expand/collapse" instead of using `aria-expanded`
3. Users can navigate into section children

Expected:
1. Sections are announced as buttons
2. Sections use `aria-expanded`
3. (Optional) Prevent navigation into section children

**Comment by csossi** [comment]
Verified in QA (as per instructions in 5342)
